torch/utils/data/dataloader.py

```diff
@@ -1144,15 +1153,18 @@ def __init__(self, loader):
 
             # Queue is not type-annotated
             self._data_queue = queue.Queue()  # type: ignore[var-annotated]
-            if self._pin_memory_device == "xpu":
-                current_device = torch.xpu.current_device()  # type: ignore[attr-defined]
+            current_device = -1
+            if self._pin_memory_device == "cuda":
+                current_device = torch.cuda.current_device()
+            elif self._pin_memory_device == "xpu":
+                current_device = torch.xpu.current_device()
             elif self._pin_memory_device == torch._C._get_privateuse1_backend_name():
                 custom_device_mod = getattr(
                     torch, torch._C._get_privateuse1_backend_name()
                 )
                 current_device = custom_device_mod.current_device()
-            else:
-                current_device = torch.cuda.current_device()  # choose cuda for default
+            elif self._pin_memory_device is None:
+                current_device = torch.accelerator.current_device_idx()
```

Comment:
```suggestion
            current_device = torch.accelerator.current_device_idx()
```
I think only this line is enough to cover those accelerators including `torch.cuda.current_device`, `torch.xpu.current_device`, and `torch.pruse1.current_device`. What do you think about it?

Reply:
In most cases, it's enough. Just for BC, in case explicitly passing a `pin_memory_device` argument (not equal to `current_accelerator`) where multiple accelerators exist in the same env. So we try to keep the same behavior as before.